code/datums/elements/waddling.dm

```diff
@@ -18,10 +22,16 @@
 		var/mob/living/living_moved = moved
 		if (living_moved.incapacitated() || living_moved.body_position == LYING_DOWN)
 			return
-	waddling_animation(moved)
+	waddling_animation(moved, hops)
 
-/datum/element/waddling/proc/waddling_animation(atom/movable/target)
-	animate(target, pixel_z = 4, time = 0)
-	var/prev_trans = matrix(target.transform)
-	animate(pixel_z = 0, transform = turn(target.transform, pick(-12, 0, 12)), time=2)
-	animate(pixel_z = 0, transform = prev_trans, time = 0)
+/datum/element/waddling/proc/waddling_animation(atom/movable/target, hopping = FALSE)
+	if(!hopping)
```

Comment:
With the above change to animation_style you'll want to make this a switch statement instead